File: php/testcode/xss_023.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_xss_multicontext
function xss_multicontext(BenchmarkRequest $req): BenchmarkResponse {
    $input = $req->param('search');

    $html = '<input type="text" value="' . $input . '">' // vuln-code-snippet vuln-line php_xss_multicontext
          . '<p>Results for: ' . $input . '</p>';

    return BenchmarkResponse::html($html);
}
// vuln-code-snippet end php_xss_multicontext

File: rust/testcode/xss_023.rs
//! CWE-79: User input placed inside a javascript: URI in an anchor href.

// vuln-code-snippet start testcodeXss023
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let action = req.param("action");

    let html = format!("<a href=\"javascript:{}\">Click</a>", action); // vuln-code-snippet target-line testcodeXss023

    super::shared::BenchmarkResponse::ok(&html)
}
// vuln-code-snippet end testcodeXss023
